frontend/CODE_GUIDELINES.md

```diff
@@ -0,0 +1,105 @@
+# Code Guidelines for React
+
+Hat tip to Clean Code by Robert C. Martin (Uncle Bob)
+
+## Goals
+
+- Consistency
+- Predictability
+- Maintainability
+- Readability
+
+## Style
+
+- Take time to craft great variable and function names. It's ok to be verbose. Avoid abbreviations.
+- Replace magic numbers with named constants.
+- Remove dead code aggressively. We can always recover it from version control.
+- Use vertical whitespace to aid readability.
+  - Related lines should be vertically close.
+  - A block of variable declarations should be followed by a blank line.
+  - Use a blank line above return statements.
+  - Don't stack code too tight vertically, especially `if` blocks.
+- Use Prettier/ESLint to avoid bike-shedding.
+- Prefer copy/paste to premature abstraction.
+  - Abstractions tend to reveal themselves quickly on the second or third copy/paste.
+  - It's easier to abstract code than to remove an abstraction.
+- Break switch statements into their own functions with returns rather than `break;`.
+
+## File structure
+
+- Higher-order functions go at the top of the file when possible.
+- Named exports are preferred to default exports. It's easier to control naming.
+- Inline exports are preferred to the define-then-export pattern.
+
+## Testing
+
+Flakey tests are the worst. The following patterns ensure that tests don't interact.
+
+- Prefer many `describe` blocks.
+- Setup and teardown belong in `beforeEach`/`afterEach` blocks if possible. 
+- `beforeAll`/`afterAll` can be a reasonable fallback.
+- Some setup may be required in `it` blocks, but try to minimize it
+- Prefer `uuid` strings to `testing-testing-123`-style strings.
+- Don't over-test the DOM structure. 
+  - Focus on testing functionality.
+  - Simple components need tests too, mostly to make sure we don't break their imports.
+  - Testing is as much about easy refactors as anything. We need to be able to refactor confidently.
+- Don't be afraid to instrument your code for testing purposes.
+- Get good at mocking imports.
+  - Declare your mocks variables at the top of the file with the `let mock*` pattern.
+  - `jest.mock()` calls should follow the `let mock*` declarations.
+  - Jest automatically hoists your `jest.mock()` calls to the top of the file, so avoid confusion by keeping them near the top of the file. 
+  - Avoid `mockFn.mockImplementation` and `mockFn.mockImplementationOnce`. They're not **bad**, but they're more verbose and a bit harder to read than the `let mock*` pattern.
+
+## Javascript
+
+- Prefer `const` over `let`. Avoid `var`.
+- Prefer named functions, i.e., `function useSomething(){}` over anonymous functions.
+- Arrow functions are best for one-liners.
```

Comment:
I'm typically team arrow function in all cases I can

Reply:
We are on different sides of a holy war. I need to pick your brain on this.